drivers/input/input_gpio_keys.c

```diff
@@ -106,6 +106,14 @@ static __maybe_unused void gpio_keys_change_deferred(struct k_work *work)
 	const struct gpio_keys_config *cfg = dev->config;
 	int key_index = pin_data - (struct gpio_keys_pin_data *)cfg->pin_data;
 
+#ifdef CONFIG_PM_DEVICE
+	struct gpio_keys_data *data = dev->data;
+
+	if (atomic_get(&data->suspended) == 1) {
+		return;
+	}
+#endif
```

Comment:
hmm why would this code run if device is suspended?

Reply:
Well I bumped into this because apparently some gpio controllers don't support `GPIO_DISCONNECTED` and the thing fails half way, but that's a different problem and needs its own thing :-). What I was thinking is that this is a delayable handler, so there may be a chance that one of these was scheduled before suspend or gets scheduled while suspend is in the loop of disabling all the interrupts, so if we exit early here we should be safe from the moment that the suspend sequence started.